kernel/kernel-api/src/test/scala/io/delta/kernel/internal/metadatadomain/JsonMetadataDomainSuite.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.metadatadomain
+
+import io.delta.kernel.internal.rowtracking.RowTrackingMetadataDomain
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.util.Optional
+
+class JsonMetadataDomainSuite extends AnyFunSuite {
+
+  test("JsonMetadataDomain can serialize/deserialize Optional fields") {
```

Comment:
could you also add a trivial test for the RowTrackingMetadataDomain class? this test is testing a _test class_, which is useful, but I'd also like to test the actual production class.